criteria/common/src/org/immutables/criteria/constraints/EmptyExpression.java

```diff
@@ -0,0 +1,20 @@
+package org.immutables.criteria.constraints;
+
+import javax.annotation.Nullable;
+
+/**
+ * Sentinel for a null expression.
+ */
+final class EmptyExpression<T> implements Expression<T> {
+
+  static final EmptyExpression INSTANCE = new EmptyExpression();
+
+  private EmptyExpression() {}
+
+  @Nullable
+  @Override
+  public <R> R accept(ExpressionVisitor<R> visitor) {
+    throw new UnsupportedOperationException("Can't visit empty expression");
```

Comment:
do nothing could be ok for an "empty". But I wonder the nature of it, probably, I should think of it as something more like bottom-type, "nonexpression" rather than "unit" / "empty" product.

Reply:
yes it is more a `Nil`